frontend-react/src/components/shared/Table/Table.tsx

```diff
@@ -0,0 +1,225 @@
+import classnames from "classnames";
+import sanitizeHtml from "sanitize-html";
+import { useMemo, useState } from "react";
+import { Icon } from "@trussworks/react-uswds";
+
+import { convertStringToSortableRepresentation } from "../../../utils/misc";
+
+import styles from "./Table.module.scss";
+
+const FilterOptionsEnum = {
+    NONE: "none",
+    ASC: "asc",
+    DESC: "desc",
+} as const;
+
+interface SortableTableHeaderProps {
+    columnTitle: string;
+    activeColumn: string;
+    sortOrder: string;
+    setSortOrder: (order: string) => void;
+    setActiveColumn: (column: string) => void;
+}
+
+interface RowData {
+    columnKey: string;
+    content: string;
+    modalContent?: string;
+}
+
+export interface TableProps {
+    borderless?: boolean;
+    compact?: boolean;
+    emptyState?: boolean;
+    fullWidth?: boolean;
+    scrollable?: boolean;
+    sortable?: boolean;
+    stackedStyle?: "default" | "headers";
+    striped?: boolean;
+    rowData: RowData[][];
+}
+
+const SortableTableHeader = ({
+    columnTitle,
+    activeColumn,
+    sortOrder,
+    setSortOrder,
+    setActiveColumn,
+}: SortableTableHeaderProps) => {
+    let SortIcon = Icon.SortArrow;
+    const isActive = columnTitle === activeColumn;
+    if (isActive && sortOrder === FilterOptionsEnum.ASC) {
+        SortIcon = Icon.ArrowUpward;
+    } else if (isActive && sortOrder === FilterOptionsEnum.DESC) {
+        SortIcon = Icon.ArrowDownward;
+    }
+    const isCurrentlyActiveColumn = columnTitle === activeColumn;
+
+    const handleHeaderClick = () => {
+        if (!isCurrentlyActiveColumn) {
+            // Reset active column and sort order on new column click
+            setActiveColumn(columnTitle);
+            setSortOrder(FilterOptionsEnum.ASC);
+        } else if (sortOrder === FilterOptionsEnum.NONE) {
+            // Explicitly set the proceeding sort order
+            setSortOrder(FilterOptionsEnum.ASC);
+        } else if (sortOrder === FilterOptionsEnum.ASC) {
+            setSortOrder(FilterOptionsEnum.DESC);
+        } else if (sortOrder === FilterOptionsEnum.DESC) {
+            setSortOrder(FilterOptionsEnum.NONE);
+        }
+    };
+    return (
+        <th
+            role="button"
+            tabIndex={0}
+            className={classnames({
+                "active-col-header": isActive,
+            })}
+            onClick={handleHeaderClick}
+        >
+            <div className="column-header-title-container">
+                <p>{columnTitle}</p>
+                {<SortIcon />}
+            </div>
+        </th>
+    );
+};
+
+const SortableTable = ({
+    rowData,
+    columnHeaders,
+}: {
+    rowData: RowData[][];
+    columnHeaders: string[];
+}) => {
+    const [activeColumn, setActiveColumn] = useState("");
+    const [sortOrder, setSortOrder] = useState(FilterOptionsEnum.NONE);
+    const sortedData = useMemo(() => {
```

Comment:
Minor point, but I don't think memoization would really help out too much here; if anything related to the table content or sorting changes, a full re-render would be expected anyway.